File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        
        while(head!=NULL && head->val==val){
            ListNode*del=head;
            head=head->next;
            delete del;
        }
        ListNode*curr=head;
        ListNode*prev=head;
        while(curr!=NULL){
            if(curr->val!=val){
                 prev=curr;
                 curr=curr->next;
            } else{
                prev->next=curr->next;
                curr=curr->next;
            }
        }
        return head;
    }
};

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode(0);
        ListNode futt=dummy;
        ListNode curr=head;
        while(curr!=null){
            if(curr.val==val){
                curr=curr.next;
            }else{
                dummy.next=curr;
                dummy=dummy.next;
                curr=curr.next;
            }
        }
        dummy.next=null;
        return futt.next;
    }
}